datafusion/common/src/hash_utils.rs

```diff
@@ -327,12 +359,16 @@ pub fn create_hashes<'a>(
                 array => hash_dictionary(array, random_state, hashes_buffer, rehash)?,
                 _ => unreachable!()
             }
+            DataType::Struct(_) => {
+                let array = as_struct_array(array)?;
+                hash_struct_array(array, random_state, hashes_buffer)?;
+            }
             DataType::List(_) => {
-                let array = as_list_array(array);
+                let array = as_list_array(array)?;
```

Comment:
Just a question: I'm a little confused about this change. Where changes the signature of `as_list_array`?

Reply:
There are two `as_list_array ` with different signatures from arrow-rs and datafusion.

Reply:
Thanks for your explanation!